app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -2141,4 +2143,32 @@ public void onClick(View view) {
     public enum Flag {
         RESET_DATABASE,
     }
+    public void setupNetworkBroadcastReceiver()
+    {
+        BroadcastReceiver networkStateReceiver = new BroadcastReceiver() {
+            final CoordinatorLayout coordinatorLayout = (CoordinatorLayout) findViewById(R.id.coordinatorLayout);
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                Log.d("Network Listener", "Network Type Changed");
+                boolean isConnected = intent.getBooleanExtra(ConnectivityManager.EXTRA_NO_CONNECTIVITY, false);
+                if(isConnected){
+                    //Disable fab when there is no connectivity
+                    fab.setEnabled(false);
```

Comment:
This is how it currently is working @kunall17  Please suggest any changes that need to be added.

![g_20170123_0106563](https://cloud.githubusercontent.com/assets/13676490/22249952/19107d6e-e26b-11e6-96bf-c0c9834130d8.gif)

The displaying of the snackbar at the bottim of the screen takes the chatbox out of the view and as soon as the network connection is reestablished the text that the user had entered is still present.